### ISSUE TITLE
[ci] Try --remote_download_minimal again

### ISSUE THREAD

**Issue by jwnimmer-tri** [RobotLocomotion/drake#21121]
We tried https://github.com/RobotLocomotion/drake-ci/pull/209 but our Bazel 6.x CI wasn't ready for that yet.

Once #21119 images are deployed, we should try re-applying that patch and see if sticks this time.

**Comment by svenevs** [comment]
Was looking back into this, the breadcrumbs:

- First attempt: https://github.com/RobotLocomotion/drake-ci/pull/209
- Reverted: https://github.com/RobotLocomotion/drake-ci/pull/210
    - Slack discussion: https://drakedevelopers.slack.com/archives/C270MN28G/p1679539645270689

Recent activity:

- https://github.com/RobotLocomotion/drake-ci/pull/266
- https://github.com/RobotLocomotion/drake-ci/pull/267

My understanding is we want to change

```diff
- --remote_download_outputs=all
+ --remote_download_outputs=minimal
```

But [`--remote_download_output` also has `toplevel`](https://bazel.build/reference/command-line-reference#flag--remote_download_outputs), which may possibly be a better choice.

Regardless of `minimal` or `toplevel`, I'm not actually sure I understand what the failure criteria is (the job logs linked in slack are gone).  Do we just YOLO it and if builds start breaking revert?

**Comment by svenevs** [comment]
> Do we just YOLO it and if builds start breaking revert?

Clarification: in https://github.com/RobotLocomotion/drake-ci/pull/209 you can get rough stats, and assuming the test jobs launch for a couple of distros, that would make the `drake-ci` PR theoretically ready to merge.  However, I don't have a record of what failed after we merged it (and therefore had to revert).

**Comment by jwnimmer-tri** [comment]
The default in Bazel 7.x was `toplevel`, which is what defeated us.
I think if toplevel didn't work, then "minimal" will also not work.

For now, we should probably put this on hold again until after Bazel 7.2 (which should be soonish).

From https://github.com/bazelbuild/bazel/issues/20161, we might need to use both `--experimental_remote_cache_eviction_retries` and `--experimental_remote_cache_lease_extension`, or else figure out why replaying the build action isn't working.

**Comment by jwnimmer-tri** [comment]
After #22066 lands, we should let CI bake for about a ~week to make sure things are stable with 7.4.0, and after that we should try this again.

By "this", I mean landing https://github.com/RobotLocomotion/drake-ci/pull/209 again and watching for fallout.  Remember that the fallout is typically several days later after merging, so we should land it on maybe a Monday and watch for PR flakes the rest of that week.

**Comment by jwnimmer-tri** [comment]
We had to revert this again.

**Comment by jwnimmer-tri** [comment]
We should try again now with Bazel 8.1.1.  (That means upgrading to use it, first.)

**Comment by tyler-yankee** [comment]
We upgraded to use Bazel 8.1.1 in https://github.com/RobotLocomotion/drake/pull/22690, so let's try again.

**Comment by jwnimmer-tri** [comment]
I'd say upgrade to 8.2.1 first (which we should do anyway on master).

**Comment by tyler-yankee** [comment]
As we've tried this through multiple (major) versions of Bazel now, perhaps better to close as not planned?

**Comment by jwnimmer-tri** [comment]
Once https://github.com/RobotLocomotion/drake/issues/21255#issuecomment-4225523576 lands and is baked for several weeks, let's try this again.

**Comment by jwnimmer-tri** [comment]
Might be worth waiting for Bazel 9.2.0 -- it has even more fixes.

**Comment by tyler-yankee** [comment]
Farmed out from #21255: In addition to `--remote_download_minimal`, we should also try `--rewind_lost_inputs`.

**Comment by jwnimmer-tri** [comment]
This is ready to try again -- we're up to Bazel 9.2.0 as of now.

**Comment by tyler-yankee** [comment]
https://github.com/RobotLocomotion/drake/issues/21121#issuecomment-4836115349

We already have `rewind_lost_inputs`, robotlocomotion/drake-ci#436. (I apparently forgot about it because it didn't have to get reverted, but also didn't seem to affect anything, and we hadn't seen "Missing digest" in a while anyways.) That leaves the third try of `--remote_download_minimal` before we call this issue fixed or won't fix.

**Comment by tyler-yankee** [comment]
Fix merged, so now we watch for the fall out :|

**Comment by jwnimmer-tri** [comment]
We've landed the CI change!  We'll see how it goes.

**Comment by tyler-yankee** [comment]
For the record we had to revert this again: robotlocomotion/drake-ci#446.

**Comment by jwnimmer-tri** [comment]
I'll put it back into the queue for Bazel 9.3.0 re-testing, in >= October.
